docs/source/app-dev/command-deduplication.rst

```diff
@@ -0,0 +1,364 @@
+     obtain a recent offset on the completion stream ``OFF1``, say its current end.
+     (Analogous to :ref:`step 2 above <dedup-bounded-step-offset>`)
+
+#. Submit the command with the following parameters (analogous to :ref:`Step 3 above <dedup-bounded-step-submit>` except for the deduplication period):
+
+   - Set the :ref:`command ID <<com.daml.ledger.api.v1.Commands.command_id>>` to the chosen command ID from :ref:`Step 1 <dedup-bounded-step-command-id>`.
+
+   - Set the :ref:`deduplication offset <com.daml.ledger.api.v1.Commands.deduplication_offset>` to ``OFF0``.
+
+   - Set the :ref:`submission ID <com.daml.ledger.api.v1.Commands.submission_id>` to a fresh value, e.g., a random UUID.
+
+   - Set the timeout (gRPC deadline) to the expected submission processing time (Command Service) or submission hand-off time (Command Submission Service).
+
+#. Wait until the RPC call returns a response.
+   
+   - Status codes other than ``OK`` should be handled according to :ref:`error handling <dedup-bounded-error-handling>`.
+
+   - When you use the :ref:`Command Service <command-service>` and the response carries the status code ``OK``, the ledger change took place.
+     You can report success.
+     The response contains a :ref:`completion offset <com.daml.ledger.api.v1.SubmitAndWaitForTransactionIdResponse.completion_offset>` that you can use in :ref:`Step 2 <dedup-unbounded-step-dedup-offset>` of later submissions.
+     
+   - When you use the :ref:`Command Submission Service <command-submission-service>`,
+     subscribe with the :ref:`Command Completion Service <command-submission-service>` for completions for ``actAs`` from ``OFF1`` (exclusive) until you see a completion event for the change ID and the submission ID chosen in :ref:`step 3 <dedup-bounded-step-submit>`.
+     If the completion’s status is ``OK``, the ledger change took place and you can report success.
+     Other status codes should be handled according to :ref:`error handling <dedup-bounded-error-handling>`.
+
+.. _dedup-unbounded-error-handling:
+
+Error handling
+--------------
+
+The same as :ref:`for known bounds <dedup-bounded-error-handling>`, except that the former retry from :ref:`Step 2 <dedup-bounded-step-offset>` becomes retry from :ref:`Step 3 <dedup-unbounded-step-offset>`.
+
+
+Failure scenarios
+-----------------
+
+The above strategy can fail in the following scenarios:
+
+#. No success within the supported deduplication period
+   
+   When the application receives a :ref:`INVALID_DEDUPLICATION_PERIOD <error_code_INVALID_DEDUPLICATION_PERIOD>` error, it cannot achieve exactly once execution any more within the originally intended deduplication period.
+
+
+#. Unacceptable changes cause infinite retries
+
+   You need business workflow knowledge to decide that retrying does not make sense any more.
+   Of course, you can always stop retrying and accept that you do not know the outcome for sure.
+
+
+
+
+.. todo:: 
```

Comment:
Do you want to add this as part of a separate PR?

Reply:
AFAIK this is not yet implemented in the JSON API server. So there's no point yet in trying to document it. Whether that's going to be me or the person that implements it in the JSON API has not yet been decided.